### PR TITLE
Provide examples of setting curl timeouts

### PR DESCRIPTION
**Pull request by sarahg**
**Note:** Please fill out the PR Template to ensure proper processing.

## Summary

**[Errors and Server Responses](https://pantheon.io/docs/errors-and-server-responses#external-web-service-calls)** - Provide examples of setting curl timeouts.

## Remaining Work
- [x] Tech review
- [x] Copy review

## Post Launch

- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6239](https://pr-6239--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `45` | `43` | `-2` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-6239` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25899/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25899/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6239--pantheon-docs.my.pantheonfrontend.website/docs)

File: source/content/errors-and-server-responses.md [hunk 1]
```diff
@@ -112,6 +112,34 @@ It is not uncommon for API or web-service integration extensions (plugins or mod
...
+```php
+curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+```
+
+Set a 10 second timeout on an external request made with Drupal 7's `drupal_http_request` function:
+
+```php
+drupal_http_request($url, $options = array('timeout' => 10))
+```
```
Comment (dficker) [inline source/content/errors-and-server-responses.md:128]: First, this should probably end with a `;` like all the other lines.

Also, I am thinking that one of these would be more clear?

```php
drupal_http_request($url, array('timeout' => 10));
````

or

```php
$options = array('timeout' => 10);
drupal_http_request($url, $options);
```

File: source/content/errors-and-server-responses.md [hunk 2]
```diff
@@ -112,6 +112,34 @@ It is not uncommon for API or web-service integration extensions (plugins or mod
...
+
+```php
+$settings['http_client_config']['timeout'] = 10;
+```
+
+Or for an individual request like this:
+
+```php
+$client = \Drupal::httpClient(['base_url' => 'https://example.com/api']);
+$client->request('GET', $url,['timeout' => 10]);
```
Comment (dficker) [inline source/content/errors-and-server-responses.md:140]: I think a space would be helpful here?

```php
$client->request('GET', $url, ['timeout' => 10]);
```

File: source/content/errors-and-server-responses.md [hunk 3]
```diff
@@ -112,6 +112,34 @@ It is not uncommon for API or web-service integration extensions (plugins or mod
...
+```
+
```
Comment (dficker) [inline source/content/errors-and-server-responses.md:142]: @jocastaneda proposed this as well for WordPress specifically. These would be added to a custom plugin or the theme's functions.php file.

```php
add_filter( 'http_request_args', 'pantheon_http_request_args', 100, 1 );
function pantheon_http_request_args( $r )
{
    $r['timeout'] = 10;
    return $r;
}
​
add_action( 'http_api_curl', 'pantheon_http_api_curl', 100, 1 );
function pantheon_http_api_curl( $handle )
{
    curl_setopt( $handle, CURLOPT_CONNECTTIMEOUT, 10 );
    curl_setopt( $handle, CURLOPT_TIMEOUT, 10 );
}
```

Related Docs for these:
https://developer.wordpress.org/reference/hooks/http_request_args/
https://developer.wordpress.org/reference/hooks/http_api_curl/

**Review by dficker — COMMENTED** [review]
Here's some thoughts on the code. I think most of these would work, but here's a few things to potentially change. I can make a commit to add these if desired.

**Comment by sarahg** [comment]
Thanks @dficker! I'm +1 on all of these suggestions. Great idea adding the WP filter/action too. I'll push another commit shortly with these edits.

**Comment by sarahg** [comment]
@alexfornuto Ready for a copy review on this one.

**Review by EdwardAngert — APPROVED** [review]

**Pull request merged**